src/OpenTelemetry.AspNetCore/OpenTelemetryServicesExtensions.cs

```diff
@@ -0,0 +1,90 @@
+﻿// <copyright file="OpenTelemetryServicesExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+using OpenTelemetry.Instrumentation.AspNetCore;
+using OpenTelemetry.Instrumentation.Dependencies;
+using OpenTelemetry.Trace.Configuration;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up OpenTelemetry services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class OpenTelemetryServicesExtensions
+    {
+        /// <summary>
+        /// Adds OpenTelemetry to the specified <see cref="IServiceCollection" /> with automatic data collection.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add OpenTelemetry to.</param>
+        /// <param name="openTelemetryOptions">The <see cref="OpenTelemetryOptions" /> to configure.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddOpenTelemetryWithInstrumentation(
+            this IServiceCollection services,
+            OpenTelemetryOptions openTelemetryOptions = null)
```

Comment:
Should we do the IOptions<OTelOptions> way? Most of the time users want to configure OpenTelemetryOptions using config.

Reply:
just added:
```
/// <summary>
        /// Adds OpenTelemetry to the specified <see cref="IServiceCollection" /> with automatic data collection.
        /// </summary>
        /// <param name="services">The <see cref="IServiceCollection" /> to add OpenTelemetry to.</param>
        /// <param name="option">The <see cref="IOptions{OpenTelemetryOptions}" /> to configure.</param>
        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
        public static IServiceCollection AddOpenTelemetryWithInstrumentation(
            this IServiceCollection services,
            IOptions<OpenTelemetryOptions> option = null)
```

Can you check if the option cref is right?

Thank you!

Reply:
Discussing offline. Eddy will modify the samples to use the new option, and then we can comment more on how much improvement an Asp.Net core special package can bring. We may start with absolute minimum with no ability to customize - and then later improve. So that we won't have to change a public API after shipped.